### PR TITLE
Deemphasize mpl_toolkits in API docs.

### PR DESCRIPTION
**Pull request by anntzer**
The docs should present mpl_toolkits as more or less regular modules,
that just happen to live in a different toplevel package for historical
reasons.  So just concatenate them to the end of the main module list.
(This is particularly in preparation of the new website redesign, where
the separate "Extra Toolkits" grouping is really too conspicuous on the
front page.)

Remove the toolkits/index page (documenting mplot3d and
axis_grid/axisartist ("ag/aa") together doesn't make much sense except
for this historical grouping), merging the mplot3d intro blurb together
with the one previously at mplot3d/index.rst; the ag/aa blurbs were
just included from the ag/aa API docs themselves and therefore don't
need to be kept.

The mplot3d intro blurb itself can go to the top of the mplot3d API
docs, rather than needing a separate page.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: doc/api/toolkits/mplot3d.rst [hunk 1]
```diff
@@ -1,3 +1,34 @@
+.. _toolkit_mplot3d-index:
+.. currentmodule:: mpl_toolkits.mplot3d
+
+*******
+mplot3d
+*******
+
+The mplot3d toolkit adds simple 3D plotting capabilities (scatter, surface,
+line, mesh, etc.) to Matplotlib by supplying an axes object that can create
```
Comment (story645) [inline doc/api/toolkits/mplot3d.rst:9]:
```suggestion
line, mesh, etc.) to Matplotlib by supplying an Axes object that can create
```

File: doc/api/toolkits/mplot3d.rst [hunk 2]
```diff
@@ -1,3 +1,34 @@
...
+more information.
+
+.. image:: /_static/demo_mplot3d.png
+   :align: center
+
+The interactive backends also provide the ability to rotate and zoom
+the 3D scene. One can rotate the 3D scene by simply clicking-and-dragging
+the scene. Zooming is done by right-clicking the scene and dragging the
+mouse up and down. Note that one does not use the zoom button like one
+would use for regular 2D plots.
```
Comment (story645) [inline doc/api/toolkits/mplot3d.rst:25]: This is unclear - is it tied to the sentence before it?

File: doc/api/toolkits/mplot3d.rst (hunk 1 above)
Comment (anntzer) [reply doc/api/toolkits/mplot3d.rst:9]: sure

File: doc/api/toolkits/mplot3d.rst (hunk 2 above)
Comment (anntzer) [reply doc/api/toolkits/mplot3d.rst:25]: reworded

File: doc/api/toolkits/mplot3d.rst [hunk 3]
```diff
@@ -1,3 +1,33 @@
+.. _toolkit_mplot3d-index:
+.. currentmodule:: mpl_toolkits.mplot3d
+
+*******
+mplot3d
```
Comment (timhoffm) [inline doc/api/toolkits/mplot3d.rst:5]: This sectioning looks awkward in the module list at https://63304-1385122-gh.circle-artifacts.com/0/doc/build/html/api/index.html

You need to:
- not use two top-level sections. (The next section should be one lebvel lower)
- Format this section analogous to the other API ReST section headers.

Comment (anntzer) [reply doc/api/toolkits/mplot3d.rst]: thanks, fixed

**Review by timhoffm — APPROVED** [review]

**Pull request merged**